pkgs/tools/misc/wit-bindgen/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "wit-bindgen";
+  version = "0.4.0";
+
+  src = fetchFromGitHub {
+    owner = "bytecodealliance";
+    repo = "wit-bindgen";
+    rev = "wit-bindgen-cli-${version}";
+    hash = "sha256-OLBuzYAeUaJrn9cUqw6nStE468TqTUXeUnfkdMO0D3w=";
+  };
+
+  cargoHash = "sha256-blaSgQZweDmkiU0Tck9qmIgpQGDZhgxb1+hs6a4D6Qg=";
+
+  # Some tests fail because they need network access to install the `wasm32-unknown-unknown` target.
```

Comment:
We can skip specific tests with `checkFlags`

Reply:
Hi @NickCao, thanks for the review.
Actually, most of the test cases in `wit-bindgen` need `wasm32-unknown-unknown`.
Since `wit-bindgen` is a generator for generating bindings, it needs `tinygo`, `wasi-sdk`, `teavm` for running test cases. But some of the above dependencies could not be found in `nixpkgs`.

Reply:
Maybe nothing for now. We surely can have better support for wasm in nixpkgs later on.